src/main/java/com/minecolonies/structures/client/TemplateRenderHandler.java

```diff
@@ -57,10 +125,78 @@ public void draw(final Template template, final Rotation rotation, final Mirror
 
                 return tessellator;
             }).draw(rotation, mirror, drawingOffset, TemplateUtils.getPrimaryBlockOffset(template));
+
+            if (tileEntities.isEmpty())
+            {
+                tileEntities = template.blocks.stream().filter(blockInfo -> blockInfo.tileentityData != null).map(b -> constructTileEntities(b, pos)).filter(Objects::nonNull).collect(Collectors.toList());
+            }
+            renderTileEntities(partialTicks, pos);
         }
         catch (ExecutionException e)
         {
             Log.getLogger().error(e);
         }
     }
+
+    @Nullable
+    private TileEntity constructTileEntities(final Template.BlockInfo info, final BlockPos pos)
+    {
+        final TileEntity entity = TileEntity.create(null, info.tileentityData);
+        if (!(entity instanceof TileEntityBanner))
+        {
+            return null;
+        }
+
+        entity.setPos(info.pos.add(pos));
+        return entity;
+    }
+
+    /**
+     * Render the list of tileEntities.
+     * @param partialTicks the partial ticks.
+     * @param pos the position.
+     */
+    private void renderTileEntities(final float partialTicks, final BlockPos pos)
+    {
+        tileEntities.forEach(tileEntity -> TileEntityRendererDispatcher.instance.render(tileEntity, partialTicks, 0));
+    }
+
+    /**
+     * Render a template at a list of points.
+     *
+     * @param points       the points to render it at.
+     * @param partialTicks the partial ticks.
+     * @param template the template.
+     */
+    public void drawTemplateAtListOfPositions(final List<BlockPos> points, final float partialTicks, final Template template)
+    {
+        if (points.isEmpty())
+        {
+            return;
+        }
+        final EntityPlayer perspectiveEntity = Minecraft.getMinecraft().player;
+        final double interpolatedEntityPosX = perspectiveEntity.lastTickPosX + (perspectiveEntity.posX - perspectiveEntity.lastTickPosX) * partialTicks;
+        final double interpolatedEntityPosY = perspectiveEntity.lastTickPosY + (perspectiveEntity.posY - perspectiveEntity.lastTickPosY) * partialTicks;
+        final double interpolatedEntityPosZ = perspectiveEntity.lastTickPosZ + (perspectiveEntity.posZ - perspectiveEntity.lastTickPosZ) * partialTicks;
+
+        if (template != this.template)
+        {
+            this.template = template;
```

Comment:
What is the reason for caching this? Isn't this method called every tick after draw anyway?